### PR TITLE
Block Library: remove uses of wrapperClassName on RichText

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Since #17607, RichText no longer has any wrapper elements if the `wrapperClassName` prop is not used. Some blocks still use this prop though, which I'd like to deprecate. First, we need to adjust all blocks and check if the styling is still good.

## How has this been tested?

Check the following blocks:

* Heading
* Preformatted
* Verse
* List
* File
* Search
* Pullquote

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/block-library/src/button/editor.scss [hunk 1]
```diff
@@ -20,11 +20,6 @@
 		cursor: text;
 	}
 
-	// Don't let placeholder expand parent width.
-	.block-editor-rich-text {
```
Comment (ellatrix) [inline packages/block-library/src/button/editor.scss]: This is dead code in `master`, and I don't see any regressions.

File: packages/block-library/src/heading/edit.js [hunk 2]
```diff
@@ -78,7 +78,6 @@ function HeadingEdit( {
 			</InspectorControls>
 			<RichText
 				identifier="content"
-				wrapperClassName="wp-block-heading"
```
Comment (ellatrix) [inline packages/block-library/src/heading/edit.js]: This class would be duplicated by the `className` prop.

File: packages/block-library/src/list/editor.scss [hunk 3]
```diff
@@ -1,5 +1,5 @@
-.editor-styles-wrapper .block-library-list ul,
-.editor-styles-wrapper .block-library-list ol {
+.editor-styles-wrapper div[data-type="core/list"] ul,
+.editor-styles-wrapper div[data-type="core/list"] ol {
```
Comment (ellatrix) [inline packages/block-library/src/list/editor.scss]: Ideally we should remove these styles... They make it harder for themes to style the list block in the editor. Anything stopping us from removing it?

File: packages/block-library/src/preformatted/theme.scss [hunk 4]
```diff
@@ -1,12 +0,0 @@
-.wp-block-preformatted {
-	pre {
```
Comment (ellatrix) [inline packages/block-library/src/preformatted/theme.scss]: This is dead code in master. There's no nested `pre` element on the front-end.

File: packages/block-library/src/button/editor.scss (hunk 1 above)
Comment (jasmussen) [reply packages/block-library/src/button/editor.scss]: Judging by the comment and in my testing, it is indeed not needed.

File: packages/block-library/src/list/editor.scss (hunk 3 above)
Comment (jasmussen) [reply packages/block-library/src/list/editor.scss]: Agree, we should remove them. However what's stopping us is, once again, the dreaded CSS bleed. Bot ul, ol and li are separately styled in an unscoped fashion by wp-admin itself. 

The two best fixes are to a) use Shadow DOM, and b) scope the wp-admin styles. We should do both long term.

Near term all we can do is reduce the specificity of these. 

It is worth noting that these are in `editor.scss`, so at least they won't affect the theme.

**Review by jasmussen — APPROVED** [review]
In my testing, this works well. I couldn't break it testing the things mentioned in 3 themes. 

I would love a sanity check by someone else on the code, but just on the testing steps, this seems good.

File: packages/block-library/src/list/editor.scss (hunk 3 above)
Comment (ellatrix) [reply packages/block-library/src/list/editor.scss]: @jasmussen I still don't get why the admin css matters if the themes provides something like

```css
.editor-styles-wrapper ul,
.editor-styles-wrapper ol {
  margin: ...;
  padding: ...;
}
```

Currently `master` will override the theme styles? Can't I just remove the extra specificity here so it properly acts as default styling.

**Comment by ellatrix** [comment]
> I would love a sanity check by someone else on the code, but just on the testing steps, this seems good.

There's not really much code changes, other than some prop removals and CSS changes or removals, so if everything is looking good, this should be good. Let's merge and fix any issues if something turns out broken.

**Pull request merged**